server/graph_service/dto/retrieve.py

```diff
@@ -0,0 +1,48 @@
+from datetime import datetime
+from typing import Literal
+
+from pydantic import BaseModel, Field
+
+from graph_service.dto.common import Message
+
+
+class SearchQuery(BaseModel):
+    group_id: str = Field(..., description='The group id of the memory to get')
+    query: str
+    max_facts: int = Field(default=10, description='The maximum number of facts to retrieve')
+    search_type: Literal['facts', 'user_centered_facts'] = Field(
+        default='facts', description='The type of search to perform'
+    )
+
+
+class FactResult(BaseModel):
```

Comment:
What's the advantage of these new data types rather than using the graphiti types?